src/main/java/xades4j/production/SignerBES.java

```diff
@@ -244,14 +248,27 @@ public final XadesSignatureResult sign(
             // with its value set to: http://uri.etsi.org/01903#SignedProperties."
 
             String digestAlgUri = algorithmsProvider.getDigestAlgorithmForDataObjsReferences();
-            if (null == digestAlgUri)
+            if (StringUtils.isNullOrEmptyString(digestAlgUri))
             {
                 throw new NullPointerException("Digest algorithm URI not provided");
             }
 
+            // Use same canonicalization URI as specified in the ds:CanonicalizationMethod for Signature.
+            String canonAlg = this.algorithmsProvider.getCanonicalizationAlgorithmForSignature().getUri();
+
             try
             {
-                signature.addDocument('#' + signedPropsId, null, digestAlgUri, null, QualifyingProperty.SIGNED_PROPS_TYPE_URI);
+                Transforms transforms = null;
+
+                if (!StringUtils.isNullOrEmptyString(canonAlg))
+                {
+                    // HACK: since we're not using Canonicalizer, do a quick check to ensure
+                    // that 'c14n' refers to a configured C14N algorithm.
+                    transforms = new Transforms(signatureDocument);
+                    transforms.addTransform(Canonicalizer.getInstance(canonAlg).getURI());

```

Comment:
Hi Luis,

Thanks for pointing this out! Based on your comments I made changes to the addTransform(..) method. Now, I was doubtful of what were those possible parameters and made some changes based on my interpretation. Can you please double check if I made the appropriate changes in: 

- https://github.com/luisgoncalves/xades4j/pull/111/commits/c6bd19eff6eea4238918bf84d4ecb64c2bd59762#diff-f8d676ba65712f301896cbc66c5bc987R85 
- https://github.com/luisgoncalves/xades4j/pull/111/commits/c6bd19eff6eea4238918bf84d4ecb64c2bd59762#diff-b5de685d575fcc090764640103b267d2R257

Let me know if this was your interpretation as well.